src/Microsoft.Crank.PullRequestBot/Program.cs

```diff
@@ -563,79 +563,101 @@ private static async Task<IEnumerable<Result>> RunBenchmark(Command command)
             // Workspace ends with path separator
             workspace = workspace.TrimEnd('\\', '/') + Path.DirectorySeparatorChar;
 
-            foreach (var run in runs)
-            {
-                File.Delete(Path.Combine(workspace, BaseFilename));
-                File.Delete(Path.Combine(workspace, HeadFilename));
+            var cloneUrl = command.PullRequest.Base.Repository.CloneUrl; // "https://github.com/dotnet/aspnetcore.git";
+            var folder = command.PullRequest.Base.Repository.Name; // $"aspnetcore"; // 
+            var baseBranch = command.PullRequest.Base.Ref; // "main"; // 
+            var prNumber = command.PullRequest.Number; // 39463;
 
-                var benchmark = _configuration.Benchmarks[run.Benchmark];
-                var profile = _configuration.Profiles[run.Profile];
-                var buildArguments = String.Join(" ", command.Components.Select(b => _configuration.Components[b].Arguments));
-                var cloneUrl = command.PullRequest.Base.Repository.CloneUrl; // "https://github.com/dotnet/aspnetcore.git";
-                var folder = command.PullRequest.Base.Repository.Name; // $"aspnetcore"; // 
-                var baseBranch = command.PullRequest.Base.Ref; // "main"; // 
-                var prNumber = command.PullRequest.Number; // 39463;
+            // Compute a unique clone folder name
+            var counter = 1;
+            while (Directory.Exists(Path.Combine(workspace, folder))) { folder = command.PullRequest.Base.Repository.Name + counter++; }
 
-                // Compute a unique clone folder name
-                var counter = 1;
-                while (Directory.Exists(Path.Combine(workspace, folder))) { folder = command.PullRequest.Base.Repository.Name + counter++; }
+            var cloneFolder = Path.Combine(workspace, folder);
 
-                var cloneFolder = Path.Combine(workspace, folder);
+            var buildCommands = command.Components.SelectMany(b => _configuration.Components[b].Script.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries));
 
-                var buildCommands = command.Components.SelectMany(b => _configuration.Components[b].Script.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries));
+            // 1- Build the base branch once
+            // 2- Run all benchmarks on all profiles
+            // 3- Build the PR branch once
```

Comment:
Maybe add a step here about how we merge the PR branch onto main.

Reply:
Added